SECEdgar/filings/cik.py

```diff
@@ -1,67 +1,26 @@
-from SECEdgar.base import _EDGARBase
-from SECEdgar.utils.exceptions import CIKError
+from SECEdgar.filings.cik_validator import CIKValidator
 
 
-class CIK(_EDGARBase):
+class CIK(object):
     """
-    Class to get and validate CIK by ticker.
+    Validates CIKs (Central Index Keys) by ticker.
 
     Attributes:
-        lookup (Union[str, list]): Ticker or list of tickers.
+        lookup (Union[str, list]): Ticker, company name, or list of tickers and/or company names.
 
     .. versionadded:: 0.1.5
     """
-    def __init__(self, lookup, **kwargs):
+
+    def __init__(self, lookups, **kwargs):
         super(CIK, self).__init__(**kwargs)
-        self._lookup = lookup
-        self._cik = self._get_all_ciks()
-        self.params.update({'action': 'getcompany'})
+        self._validator = CIKValidator(lookups)
```

Comment:
Why a object needs to validate itself during initialisation? Here CIK class is trying to validate itself that whether it is a valid object or not. 

This make sense if we have some small regex or string matching to do but if we need to make network request this will overly complicate the instantiation of the object. Why not let the parent do the validation.

or initialise the object with just lookups and provide an API which will validate later.

Reply:
What class will validate and when? Would we add a method to CIK to be called by classes that take in a CIK object?
```
class CIK:
# other code

    def validate(self):
        # Validate given lookups here

```